tests/data/test_read_and_edit_bucket_csv_files.py

```diff
@@ -0,0 +1,44 @@
+import os
+import pytest
+import numpy as np
+import pandas as pd
+from glambie.data.read_and_edit_bucket_csv_files import download_csvs_from_bucket, check_glambie_submission_for_errors
+
+
+@pytest.fixture()
+def example_file_check_dataframe():
+    return pd.DataFrame.from_dict({'local_filepath': ['dummy', 'dummy'],
+                                   'date_check_satisfied': np.nan,
+                                   'nodata_check_satisfied': np.nan})
+
+
+def test_download_csvs_from_bucket(tmp_path):
+
+    test_prefix = 'acs_altimetry'
+    local_data_directory_path = str(tmp_path)  # temp folder for test of download
+    downloaded_files = download_csvs_from_bucket(local_data_directory_path, region_prefix=test_prefix)
```

Comment:
here I think you'll need to patch the google bucket client call to avoid actually downloading any data, instead check the calls to the client are with the inputs you expect them to be

Reply:
Agreed - you don't want to test the google client or the bucket structure, just your code. It'll be a little finnicky in this case but there's a good reference in here on how to mock some parts of the google client and the objects it works with. Happy to work through an example together. e.g. https://github.com/earthwave/ew_tile_server/blob/main/tests/test_google_bucket_source.py

Reply:
Done :)